### ISSUE TITLE
word2vec's reset_from() raise an AttributeError

### ISSUE THREAD

**Issue by robotcator** [RaRe-Technologies/gensim#1242]
#### Description
The word2vec module's function reset_from(other_model) thrown an AttributeError, the sample code are below.

#### Steps/Code/Corpus to Reproduce

```
from gensim.models import word2vec

sentences = [
    ['human', 'interface', 'computer'],
    ['survey', 'user', 'computer', 'system', 'response', 'time'],
    ['eps', 'user', 'interface', 'system'],
    ['system', 'human', 'system', 'eps'],
    ['user', 'response', 'time'],
    ['trees'],
    ['graph', 'trees'],
    ['graph', 'minors', 'trees'],
    ['graph', 'minors', 'survey']
]

models = [
    word2vec.Word2Vec(size=2, min_count=1, hs=1, negative=0),
    word2vec.Word2Vec(size=4, min_count=1, hs=1, negative=0),
]
models[0].build_vocab(sentences)
models[1].reset_from(models[0])
```

#### Actual Results

the output:
```
Traceback (most recent call last):
  File "test_reset_from.py", line 20, in <module>
    models[1].reset_from(models[0])
  File "/home/robotcator/PycharmProjects/gensim/gensim/models/word2vec.py", line 734, in reset_from
    self.wv.vocab = other_model.vocab
AttributeError: 'Word2Vec' object has no attribute 'vocab'
```

#### Versions

>>> import platform; print(platform.platform())
Linux-3.13.0-24-generic-x86_64-with-Ubuntu-14.04-trusty
>>> import sys; print("Python", sys.version)
('Python', '2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]')
>>> import numpy; print("NumPy", numpy.__version__)
('NumPy', '1.11.0')
>>> import scipy; print("SciPy", scipy.__version__)
('SciPy', '0.13.3')
>>> import gensim; print("gensim", gensim.__version__)
('gensim', '1.0.1')
>>> from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
('FAST_VERSION', 1)

#### Solutions
I have created a pull request for this issue [pr1241](https://github.com/RaRe-Technologies/gensim/pull/1241)

**Comment by gojomo** [comment]
Duplicate of #1230, fix in progress at #1234.

**Comment by tmylk** [comment]
Fixed in #1234
